src/components/forms/input-hoc.jsx

```diff
@@ -0,0 +1,20 @@
+var React = require('react');
+
+module.exports = function InputComponentMixin (Component) {
+    var InputComponent = React.createClass({
+        getDefaultProps: function () {
+            return {
+                messages: {
+                    'general.notRequired': 'Not Required'
+                }
+            };
+        },
+        render: function () {
+            return (
+                <Component help={this.props.required ? null : this.props.messages['general.notRequired']}
```

Comment:
Might be easier to read if we move the ternary into a var at the top of the `render` function.